### PR TITLE
Light refactoring of reducer

### PR DESCRIPTION
**Pull request by emilyb7**
Attempt to break down some of the functions in the reducer + adds some more detailed tests for reducer functions - #194

### CODE REVIEW

**Comment by des-des** [comment]
@emilyb7 can you merge then re-assign

File: app/components/view-goal/progress-bar.jsx [hunk 1]
```diff
@@ -1,31 +1,32 @@
 import React from 'react';
 
-const ProgressBarComponent = ({ progress, time, }) => {
+const ProgressBarComponent = ({ rating, time, }) => {
+  const score = parseInt(rating, 10);
```
Comment (des-des) [inline app/components/view-goal/progress-bar.jsx]: Parse this as it comes out of the event, so it is an int everywhere in your application. Good work with the second argument

File: app/reducers/index.js [hunk 2]
```diff
@@ -9,37 +9,54 @@ const defaultState = {
   currentGoal: {},
 };
 
+export const mapWithId = ({ goals, }, { id, }, fn) =>
+goals.map(elem =>
+  elem.id === id
+  ? fn(elem)
```
Comment (des-des) [inline app/reducers/index.js]: indentation (or maybe this one is a preference thing)

File: app/reducers/index.js [hunk 3]
```diff
@@ -9,37 +9,54 @@ const defaultState = {
...
-    newRating: {},
-  };
-  const goals = state.goals.map((goal) => goal.id === currentGoal.id ?
-    increaseUpdateCount(currentGoal) : goal);
+
+  const newRating = constructNewRating(state, time, id);
+  const currentGoal = addRatingToCurrentGoal(state, newRating);
+  const goals = state.goals.map((goal) =>
+    goal.id === currentGoal.id
+    ? increaseUpdateCount(currentGoal)
```
Comment (des-des) [inline app/reducers/index.js]: And here but am happy if this indentation was a choice you made

File: app/reducers/index.js [hunk 4]
```diff
@@ -9,37 +9,54 @@ const defaultState = {
...
+export const increaseUpdateCount = goal => {
   return { ...goal, updateCount: (goal.updateCount + 1 || 1), };
 };
 
+export const constructNewRating = ({ currentGoal, }, time, id) => {
+  return {
+    score: currentGoal.newRating.score,
+    id: id,
+    time: time,
+    comment: currentGoal. newRating.comment,
```
Comment (des-des) [inline app/reducers/index.js]: extra whitespace (does this run)

File: app/reducers/index.js [hunk 5]
```diff
@@ -9,37 +9,54 @@ const defaultState = {
...
+  };
+};
+
+export const addGoalToArray = (state, { goal, }, fn = (goal) => { return goal; }) => {
```
Comment (des-des) [inline app/reducers/index.js]: `fn = goal => goal`

Comment (des-des) [inline app/reducers/index.js]: again not a blocker

File: tests/reducer-test.js [hunk 6]
```diff
@@ -14,24 +19,109 @@ const defaultState = {
...
+    };
+  };
+  const action = { goal: goal2, };
+
+  t.equal(addGoalToArray(initialState, action).length, 2, '2 goals in array');
+  t.equal(addGoalToArray(initialState, action, fn).length, 2,
+      'still 2 goals in array'
+    );
+  t.equal(addGoalToArray(initialState, action, fn).pop().updateCount, 3,
+      'optional function applied'
```
Comment (des-des) [inline tests/reducer-test.js]: liking da optional function

File: app/reducers/index.js (hunk 4 above)
Comment (emilyb7) [inline app/reducers/index.js]: yes, it does :p

**Pull request merged**